scripts/Microsoft.DotNet.Cli.Build.Framework/TargetConditions/EnvironmentAttribute.cs

```diff
@@ -24,15 +23,22 @@ public override bool EvaluateCondition()
         {
             var actualVal = Environment.GetEnvironmentVariable(_envVar);
 
-            foreach (var expectedVal in _expectedVals)
+            if (_expectedVals.Length == 0)
             {
-                if (string.Equals(actualVal, expectedVal, StringComparison.Ordinal))
+                return !string.IsNullOrEmpty(actualVal);
+            }
+            else
```

Comment:
I talked to Michael about this offline, and he suggested using LINQ to make the code more readable. I do still prefer keeping the "else" because that seems more logical to me for this case. If using this method without specifying arguments for expected values were truly an edge case, then I would absolutely agree with not having an `else`, but I don't see it as such.